<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <meta name="description" content="None">
  
  <link rel="shortcut icon" href="img/favicon.ico">
  <title>Metavisitor - Metavisitor-2</title>
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>

  <link rel="stylesheet" href="css/theme.css" type="text/css" />
  <link rel="stylesheet" href="css/theme_extra.css" type="text/css" />
  <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
  
  <script>
    // Current page data
    var mkdocs_page_name = "Metavisitor";
    var mkdocs_page_input_path = "index.md";
    var mkdocs_page_url = null;
  </script>
  
  <script src="js/jquery-2.1.1.min.js" defer></script>
  <script src="js/modernizr-2.8.3.min.js" defer></script>
  <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
      <div class="wy-side-nav-search">
        <a href="." class="icon icon-home"> Metavisitor-2</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="./search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
	<ul class="current">
	  
          
            <li class="toctree-l1 current">
		
    <a class="current" href=".">Metavisitor</a>
    <ul class="subnav">
            
    <li class="toctree-l2"><a href="#quick-start">Quick Start</a></li>
    
        <ul>
        
            <li><a class="toctree-l3" href="#availability-of-metavisitor-tools-and-workflows">Availability of Metavisitor tools and workflows</a></li>
        
            <li><a class="toctree-l3" href="#starting-a-metavisitor-galaxy-server-from-scratch">Starting a Metavisitor Galaxy server from scratch</a></li>
        
        </ul>
    

    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="metavisitor_configure_references/">Prepare Metavisitor Galaxy instance for analyses</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="use_cases_input_data/">Prepare input data histories</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="use_case_1/">Use Cases 1-1 to 1-4</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="use_case_2/">Use Cases 2-1 and 2-2</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="use_case_3-1/">Use Case 3-1</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="use_case_3-2/">Use Case 3-2</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="use_case_3-3/">Use Case 3-3</a>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Install_Metavisitor</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="install_metavisitor/">Intro</a>
                </li>
                <li class="">
                    
    <a class="" href="metavisitor_ansible/">GalaxyKickStart</a>
                </li>
                <li class="">
                    
    <a class="" href="metavisitor_docker/">Docker</a>
                </li>
                <li class="">
                    
    <a class="" href="metavisitor_access_control/">Access and Control Metavisitor Galaxy instance</a>
                </li>
    </ul>
	    </li>
          
        </ul>
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href=".">Metavisitor-2</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href=".">Docs</a> &raquo;</li>
    
      
    
    <li>Metavisitor</li>
    <li class="wy-breadcrumbs-aside">
      
    </li>
  </ul>
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <p><img alt="metavisitor_logo" src="images/metavisitor_logo.png" /></p>
<p><a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0168397">Metavisitor</a> is
a user-friendly and adaptable software to provide biologists, clinical researchers and
possibly diagnostic clinicians with the ability to robustly detect and reconstruct viral
genomes from complex deep sequence datasets. A set of modular bioinformatic tools and
workflows was implemented as the Metavisitor package in the Galaxy framework. Using the
graphical Galaxy workflow editor, users with minimal computational skills can use existing
Metavisitor workflows or adapt them to suit specific needs by adding or modifying analysis
modules.</p>
<h1 id="quick-start">Quick Start</h1>
<p>Users who want to use Metavisitor on the <a href="https://mississippi.snv.jussieu.fr">Galaxy Mississippi Server</a>,
or got already the Metavisitor suite of tools installed in their own Galaxy server, can
jump to the next chapter <a href="use_cases_input_data">Prepare input data histories</a>.</p>
<h2 id="availability-of-metavisitor-tools-and-workflows">Availability of Metavisitor tools and workflows</h2>
<p>Metavisitor has been developed at the <a href="http://artbio.fr">ARTbio platform</a>. Its tools are
primarily available in [GitHub] (https://github.com/ARTbio/tools-artbio). Its workflows are
primarily available in this
<a href="https://github.com/ARTbio/metavisitor/tree/master/extra-files/metavisitor">metavisitor repository</a>
as well as in the <a href="https://github.com/ARTbio/GalaxyKickStart/tree/master/extra-files/metavisitor">GalaxyKickStart repository</a></p>
<h4 id="metavisitor-tools-developed-by-artbio-in-the-artbio-github">Metavisitor tools developed by ARTbio in the <a href="https://github.com/ARTbio/tools-artbio">ARTbio GitHub</a></h4>
<ul>
<li><a href="https://github.com/ARTbio/tools-artbio/tree/master/tools/blast_to_scaffold"><code>blast_to_scaffold</code></a></li>
<li><a href="https://github.com/ARTbio/tools-artbio/tree/master/tools/blastx_to_scaffold"><code>blastx_to_scaffold</code></a></li>
<li><a href="https://github.com/ARTbio/tools-artbio/tree/master/tools/cherry_pick_fasta"><code>cherry_pick_fasta</code></a></li>
<li><a href="https://github.com/ARTbio/tools-artbio/tree/master/tools/concatenate_multiple_datasets"><code>concatenate_multiple_datasets</code></a></li>
<li><a href="https://github.com/ARTbio/tools-artbio/tree/master/tools/fetch_fasta_from_ncbi"><code>fetch_fasta_from_ncbi</code></a></li>
<li><a href="https://github.com/ARTbio/tools-artbio/tree/master/tools/msp_blastparser_and_hits"><code>msp_blastparser_and_hits</code></a></li>
<li><a href="https://github.com/ARTbio/tools-artbio/tree/master/tools/msp_cap3"><code>msp_cap3</code></a></li>
<li><a href="https://github.com/ARTbio/tools-artbio/tree/master/tools/msp_fasta_tabular_converter"><code>msp_fasta_tabular_converter</code></a></li>
<li><a href="https://github.com/ARTbio/tools-artbio/tree/master/tools/msp_sr_bowtie"><code>msp_sr_bowtie</code></a></li>
<li><a href="https://github.com/ARTbio/tools-artbio/tree/master/tools/msp_oases"><code>msp_oases</code></a></li>
<li><a href="https://github.com/ARTbio/tools-artbio/tree/master/tools/small_rna_maps"><code>small_rna_maps</code></a>.</li>
<li><a href="https://github.com/ARTbio/tools-artbio/tree/master/tools/yac_clipper"><code>yac_clipper</code></a></li>
</ul>
<p>Other tools from other developers are used in the suite_metavisitor_2. These tools are
available from the <a href="https://toolshed.g2.bx.psu.edu/">main Galaxy toolshed</a>:</p>
<pre><code> name="bowtie2" owner="devteam"
 name="data_manager_bowtie2_index_builder" owner="devteam"
 name="data_manager_fetch_genome_dbkeys_all_fasta" owner="devteam"
 name="fasta_compute_length" owner="devteam"
 name="fasta_filter_by_length" owner="devteam"
 name="fastx_trimmer" owner="devteam"
 name="ncbi_blast_plus" owner="devteam"
 name="data_manager_bowtie_index_builder" owner="iuc"
 name="khmer_normalize_by_median" owner="iuc"
 name="sra_tools" owner="iuc"
 name="trinity" owner="iuc"
 name="vsearch" owner="iuc"
 name="regex_find_replace" owner="galaxyp"
 name="spades" owner="nml"
</code></pre>
<h4 id="availability-of-metavisitor-tools-and-workflows-for-galaxy-instance-administrators">Availability of Metavisitor tools and workflows for <strong>Galaxy instance administrators</strong></h4>
<p>All metavisitor tools are available in the
<a href="https://toolshed.g2.bx.psu.edu/repository/browse_repositories?sort=name&amp;operation=view_or_manage_repository&amp;f-free-text-search=metavisitor&amp;id=ca18473f5a7e691a#TODO: Change this link when the suite is published">suite_metavisitor_2</a>
Galaxy Admin can just install this suite of tools by using the <code>Install new tools</code> menu in
their Admin panel, searching for "metavisitor", and installing the <code>suite_metavisitor_2</code>
tool suite.</p>
<p>Admins can also install the tools from the <code>metavisitor_workflows</code> repository, which will
provide in addition the metavisitors workflows.</p>
<h4 id="availability-of-metavisitors-workflows-for-any-galaxy-instance-user">Availability of Metavisitors workflows for any Galaxy instance user.</h4>
<p>We have deposited the Metavisitors workflows in the
<a href="http://www.myexperiment.org/workflows #TODO: this will need to be updated too">myexperiment server</a>,
where they are searchable with "metavisitor" keyword and can be downloaded and reuploaded
to the Galaxy instance.</p>
<h2 id="starting-a-metavisitor-galaxy-server-from-scratch">Starting a Metavisitor Galaxy server from scratch</h2>
<p>In the <a href="https://artbio.github.io/metavisitor/install_metavisitor/">last section</a> of this
documentation, we provide instructions to set up Galaxy server instances <em>from scratch</em>
with <em>pre-installed</em> Metavisitor tools and workflows:</p>
<ul>
<li>Based on Ansible: see <a href="metavisitor_ansible/">Metavisitor with GalaxyKickstart (Ansible)</a></li>
<li>Based on Docker: see <a href="metavisitor_docker/">Metavitor with Docker</a></li>
</ul>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="metavisitor_configure_references/" class="btn btn-neutral float-right" title="Prepare Metavisitor Galaxy instance for analyses">Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="http://www.mkdocs.org">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" style="cursor: pointer">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
      
      
        <span style="margin-left: 15px"><a href="metavisitor_configure_references/" style="color: #fcfcfc">Next &raquo;</a></span>
      
    </span>
</div>
    <script>var base_url = '.';</script>
    <script src="js/theme.js" defer></script>
      <script src="search/main.js" defer></script>

</body>
</html>

<!--
MkDocs version : 1.0.4
Build Date UTC : 2019-07-24 12:45:23
-->
