<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  <link rel="shortcut icon" href="../img/favicon.ico">
  <title>Use Case 3-1 - Metavisitor-2</title>
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>

  <link rel="stylesheet" href="../css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../css/theme_extra.css" type="text/css" />
  <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
  
  <script>
    // Current page data
    var mkdocs_page_name = "Use Case 3-1";
    var mkdocs_page_input_path = "use_case_3-1.md";
    var mkdocs_page_url = null;
  </script>
  
  <script src="../js/jquery-2.1.1.min.js" defer></script>
  <script src="../js/modernizr-2.8.3.min.js" defer></script>
  <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
      <div class="wy-side-nav-search">
        <a href=".." class="icon icon-home"> Metavisitor-2</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
	<ul class="current">
	  
          
            <li class="toctree-l1">
		
    <a class="" href="..">Metavisitor</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../metavisitor_configure_references/">Prepare Metavisitor Galaxy instance for analyses</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../use_cases_input_data/">Prepare input data histories</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../use_case_1/">Use Cases 1-1 to 1-4</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../use_case_2/">Use Cases 2-1 and 2-2</a>
	    </li>
          
            <li class="toctree-l1 current">
		
    <a class="current" href="./">Use Case 3-1</a>
    <ul class="subnav">
            
    <li class="toctree-l2"><a href="#input-data-for-use-case-3-1">Input data for Use Case 3-1</a></li>
    

    <li class="toctree-l2"><a href="#history-for-use-case-3-1">History for Use Case 3-1</a></li>
    

    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../use_case_3-2/">Use Case 3-2</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../use_case_3-3/">Use Case 3-3</a>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Install_Metavisitor</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../install_metavisitor/">Intro</a>
                </li>
                <li class="">
                    
    <a class="" href="../metavisitor_ansible/">GalaxyKickStart</a>
                </li>
                <li class="">
                    
    <a class="" href="../metavisitor_docker/">Docker</a>
                </li>
                <li class="">
                    
    <a class="" href="../metavisitor_access_control/">Access and Control Metavisitor Galaxy instance</a>
                </li>
    </ul>
	    </li>
          
        </ul>
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="..">Metavisitor-2</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="..">Docs</a> &raquo;</li>
    
      
    
    <li>Use Case 3-1</li>
    <li class="wy-breadcrumbs-aside">
      
    </li>
  </ul>
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <p>Now that you are familiar with manipulations in Galaxy with the Use Cases 1-1 to 1-4 described in detail in the previous chapters, we will describe the other Use Case analyses more concisely. If you experience lack of skills in basic Galaxy operations (tool usage, copy of datasets, etc), do not hesitate to go back and examine the <a href="use_cases_input_data">previous chapters</a> step by step.</p>
<h2 id="input-data-for-use-case-3-1">Input data for Use Case 3-1</h2>
<p>As for the previous Use Cases 1 and 2, the first step is to collect all the input data in a history that we will name <code>Input data for Use Case 3-1</code>.</p>
<ol>
<li>
<p>Create a new history</p>
<ul>
<li>Rename this history <code>Input data for Use Case 3-1</code></li>
<li>
<p>We are going to upload 40 datasets form the EBI ENA SRP068722 :</p>
<p>Go to the upload files menu and select <code>Paste/Fetch data</code>. Copy-Paste the following table (excluding the headers):</p>
<table>
<thead>
<tr>
<th>SRR id</th>
<th>Patient id</th>
</tr>
</thead>
<tbody>
<tr>
<td>SRR3111582</td>
<td>patient 0450-318</td>
</tr>
<tr>
<td>SRR3111583</td>
<td>patient 0450-318</td>
</tr>
<tr>
<td>SRR3111584</td>
<td>patient 0450-318</td>
</tr>
<tr>
<td>SRR3111585</td>
<td>patient 0450-318</td>
</tr>
<tr>
<td>SRR3111586</td>
<td>patient 0450-318</td>
</tr>
<tr>
<td>SRR3111587</td>
<td>patient 0450-318</td>
</tr>
<tr>
<td>SRR3111588</td>
<td>patient 0387-272</td>
</tr>
<tr>
<td>SRR3111589</td>
<td>patient 0387-272</td>
</tr>
<tr>
<td>SRR3111590</td>
<td>patient 0387-272</td>
</tr>
<tr>
<td>SRR3111591</td>
<td>patient 0387-272</td>
</tr>
<tr>
<td>SRR3111592</td>
<td>patient 0387-272</td>
</tr>
<tr>
<td>SRR3111593</td>
<td>patient 0387-272</td>
</tr>
<tr>
<td>SRR3111594</td>
<td>patient 0629-453</td>
</tr>
<tr>
<td>SRR3111595</td>
<td>patient 0629-453</td>
</tr>
<tr>
<td>SRR3111596</td>
<td>patient 0629-453</td>
</tr>
<tr>
<td>SRR3111597</td>
<td>patient 0629-453</td>
</tr>
<tr>
<td>SRR3111598</td>
<td>patient 0629-453</td>
</tr>
<tr>
<td>SRR3111599</td>
<td>patient 0629-453</td>
</tr>
<tr>
<td>SRR3111600</td>
<td>patient 0444-312</td>
</tr>
<tr>
<td>SRR3111601</td>
<td>patient 0444-312</td>
</tr>
<tr>
<td>SRR3111602</td>
<td>patient 0444-312</td>
</tr>
<tr>
<td>SRR3111603</td>
<td>patient 0444-312</td>
</tr>
<tr>
<td>SRR3111604</td>
<td>patient 0500-355neg</td>
</tr>
<tr>
<td>SRR3111605</td>
<td>patient 0500-355neg</td>
</tr>
<tr>
<td>SRR3111606</td>
<td>patient 0292-xxxneg</td>
</tr>
<tr>
<td>SRR3111607</td>
<td>patient 0292-xxxneg</td>
</tr>
<tr>
<td>SRR3111608</td>
<td>patient 0394-274</td>
</tr>
<tr>
<td>SRR3111609</td>
<td>patient 0394-274</td>
</tr>
<tr>
<td>SRR3111610</td>
<td>patient 0218-162neg</td>
</tr>
<tr>
<td>SRR3111611</td>
<td>patient 0218-162neg</td>
</tr>
<tr>
<td>SRR3111612</td>
<td>patient 0311-217HIVneg</td>
</tr>
<tr>
<td>SRR3111613</td>
<td>patient 0311-217HIVneg</td>
</tr>
<tr>
<td>SRR3111614</td>
<td>patient 0440-307neg</td>
</tr>
<tr>
<td>SRR3111616</td>
<td>patient 0440-307neg</td>
</tr>
<tr>
<td>SRR3111617</td>
<td>patient 0518-370neg</td>
</tr>
<tr>
<td>SRR3111618</td>
<td>patient 0518-370neg</td>
</tr>
<tr>
<td>SRR3111619</td>
<td>patient 0560-420neg</td>
</tr>
<tr>
<td>SRR3111620</td>
<td>patient 0560-420neg</td>
</tr>
<tr>
<td>SRR3111621</td>
<td>patient 0575-419neg</td>
</tr>
<tr>
<td>SRR3111622</td>
<td>patient 0575-419neg</td>
</tr>
</tbody>
</table>
</li>
<li>
<p>Name the file "Use-Case_3-1_information" and click the <code>Start</code> button.</p>
</li>
<li>Use the tool <code>Cut columns from table</code>. In the "Cut columns field" write <code>c1</code> and make sure you select "Use-Case_3-1_information" file in the "From" field before executing. Rename the output "Use-Case_3-1_accessions".</li>
<li>Use the tool <code>Extract reads in FASTQ/A format from NCBI SRA</code>, select <code>List of SRA accession, one per line</code>from <code>select input type</code> and "Use-Case_3-1_accessions" in sra accession list. Click the <code>Execute</code> button.</li>
<li>When the tool is finished running you should have 2 new dataset collections in your history, one of them is empty. Delete the empty collection and verify that you have 40 pairs of datasets in the second collection.</li>
<li>If you are missing some sequences you'll have to re-do the steps above with only the missing identifiers. Once done, merge the collections using the tool <code>Merge Collections</code>.</li>
<li>Use the <code>Concatenate multiple datasets tail-to-head</code> tool and select "Paired collection" as type of data. Set the paired collection as input and select "Concatenate pairs of datasets" as type of concatenation. Execute the tool.</li>
<li>Rename the outputed collection to <code>SRP068722</code> and delete the previous one by clicking the <code>X</code> button and selecting "Permanently Delete Datasets".</li>
</ul>
</li>
<li>
<p>Copy the <code>vir2 nucleotide BLAST database</code> from the <code>References</code> history to the current history <code>Input data for Use Case 3-1</code>.</p>
</li>
<li>Now we still have to associate sequencing dataset coming from the same patient. We are going to use the tool <code>Tag elements from file</code> to add the patient information as metadata.<ul>
<li>Click on the <code>Tag elements from file</code> tool and select the collection "SRP068722" in "Input Collection" and "Use-Case_3-1_information" in "Tag collection elements according to this file". Execute the tool. Rename the new dataset collection <code>SRP068722_with_patient_information</code>.</li>
<li>Select the <code>Apply Rule to Collection</code> and set "SRP068722_with_patient_information" as "Input Collection". Click on the "Edit" button at the right of the form.<ul>
<li>Click the "Column" button and select <code>Add Column from Metadata</code> from the list.</li>
<li>In the "From" list select "Tags". Then click the "Apply" button.</li>
<li>Click the "Rules" button and select <code>Add / Modify Colmn Definitions</code> from the list.</li>
<li>Click the "Add Definition" button and select the <code>List identifier(s)</code> from the list.</li>
<li>In the "Select a column" list select "B" then click on <code>... Assign Another Column</code> and select "A". Click the "Apply" button.</li>
<li>Click the "Save" and execute the tool.</li>
<li>Select the <code>Concatenate multiple datasets tail-to head</code> tool. In "What type of data do you wish to concatenate?" select "Nested collection". In "Input nested collection" select "SRP068722_with_patient_information (re-organized)". Execute the tool.</li>
<li>Rename the resulting collection "patient collection".</li>
</ul>
</li>
</ul>
</li>
<li>We are done. You can now permanently delete "SRP068722_with_patient_information",  "SRP068722_with_patient_information (re-organized)" and "SRP068722". This will save you some disk space.</li>
</ol>
<h2 id="history-for-use-case-3-1">History for Use Case 3-1</h2>
<ol>
<li>Stay in the history <code>Input data for Use Case 3-1</code></li>
<li>pick the workflow <code>Metavisitor: Workflow for Use Case 3-1</code> in the workflows menu, and select the <code>run</code> option.</li>
<li>For Step 1 (Fever Patient Sequences collection), select <code>patient collection</code> (this should be already selected).</li>
<li>For Step 2, select the <code>nucleotide vir2 blast database</code> (this should also be already selected)</li>
<li>As usual, check the box <code>Send results to a new history</code>, edit the name of the new history to <code>History for Use Case 3-1</code>, and <code>Execute</code> the workflow ! Note, that for complex workflows with dataset collections in input, the actual warning that the workflow is started may take time to show up.</li>
</ol>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../use_case_3-2/" class="btn btn-neutral float-right" title="Use Case 3-2">Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href="../use_case_2/" class="btn btn-neutral" title="Use Cases 2-1 and 2-2"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="http://www.mkdocs.org">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" style="cursor: pointer">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
      
        <span><a href="../use_case_2/" style="color: #fcfcfc;">&laquo; Previous</a></span>
      
      
        <span style="margin-left: 15px"><a href="../use_case_3-2/" style="color: #fcfcfc">Next &raquo;</a></span>
      
    </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme.js" defer></script>
      <script src="../search/main.js" defer></script>

</body>
</html>
